internal/testrunner/runners/policy/policy.go

```diff
@@ -190,7 +190,14 @@ var uniqueOtelComponentIDReplace = policyEntryReplace{
 //	            - endpoints:
 //	                - https://epr.elastic.co
 //	              method: GET
-var otelComponentIDsRegexp = regexp.MustCompile(`(?m)^(?:extensions|receivers|processors|connectors|exporters):(?:\s\{\}\n|\n(?:\s{2,}.+\n)+)`)
+//	   service:
+//	      pipelines:
+//	          logs:
+//	              receivers/6b7f1379-dcb9-4ac7-b253-4df6d088b3ff:
+//	                  - httpcheck/b0f518d6-4e2d-4c5d-bda7-f9808df537b7
+//
+// The regex captures the whole section, so it can be processed line by line to replace the IDs.
+var otelComponentIDsRegexp = regexp.MustCompile(`(?m)^(?:extensions|receivers|processors|connectors|exporters|service):(?:\s\{\}\n|\n(?:\s{2,}.+\n)+)`)
```

Comment:
Included the `service` map into this regex in order to take into account these kind of definitions too to find pipelines IDs like (`metrics/<some_id>`):
```yaml
service:
  pipelines:
    logs:
      receivers/6b7f1379-dcb9-4ac7-b253-4df6d088b3ff:
        - httpcheck/b0f518d6-4e2d-4c5d-bda7-f9808df537b7
```